src/aes_decrypter.cpp

```diff
@@ -23,8 +30,21 @@ void AESDecrypter::decrypt(const AP4_UI08 *aes_key, const AP4_UI08 *aes_iv, cons
     16,
     cbc_d_block_cipher);
 
-  cbc_d_block_cipher->Process(src, dataSize, dst, aes_iv);
+  cbc_d_block_cipher->Process(src, dataSize, reinterpret_cast<AP4_UI08*>(&dst[0] + dstOffset),
```

Comment:
instead to use `cbc_d_block_cipher->Process` is possible that this method:
```c++
AP4_CbcStreamCipher cbcStreamCipher{cbc_d_block_cipher};
cbcStreamCipher.ProcessBuffer(...)
```
already do the padding check?

Reply:
Thanks for pointing this out. I tried to do this way initially but was missing passing the block cipher into the stream cipher constructor - what I came up with was just copy/paste/adjust what was in AP4_CbcStreamCipher::DecryptBuffer to remove the padding.
Looks much nicer now and cbcStreamCipher also takes care of deleting the block cipher :)

Reply:
nice! for safety better if @TermeHansen give another test

Reply:
yep, still works...

Reply:
Thank you